ui/src/components/ConnectionStatus.tsx

```diff
@@ -1,8 +1,8 @@
 import { useAPI } from "../contexts/APIProvider";
-import { useMemo } from "react";
+import { useMemo, useEffect } from "react";
 
 const ConnectionStatusIcon = () => {
-  const { connectionStatus } = useAPI();
+  const { connectionStatus, versionInfo, getVersionInfo } = useAPI();
```

Comment:
getVersionInfo() shouldn't be done as a `useEffect` here. Instead, APIProvider can fetch it on init, save it and export it. It's const that doesn't change after all.